demo.cc

```diff
@@ -576,16 +579,134 @@ class Cruncher final {
   void operator=(Cruncher&&) = delete;
 };
 
+class MixpanelUploader final {
+ public:
+  MixpanelUploader(const std::string& demo_id, const std::string& mixpanel_token)
+      : demo_id_(demo_id), mixpanel_token_(mixpanel_token) {}
+
+  inline bool Entry(std::unique_ptr<schema::Base>& entry, size_t index, size_t total) {
+    static_cast<void>(index);
+    static_cast<void>(total);
+
+    struct types {
+      typedef schema::Base base;
+      typedef std::tuple<schema::AnswerRecord> derived_list;
+      typedef bricks::rtti::RuntimeTupleDispatcher<base, derived_list> dispatcher;
+    };
+    types::dispatcher::DispatchCall(*entry, *this);
+
+    return true;
+  }
+
+  inline void Terminate() { std::cerr << '@' << demo_id_ << " MixpanelUploader is done.\n"; }
+
+  inline void operator()(schema::Base&) { /* ignore */
+  }
+
+  inline void operator()(schema::AnswerRecord& a) {
+    std::cerr << '@' << demo_id_ << " MixpanelUploader +A: " << a.uid << " `" << static_cast<int>(a.answer)
+              << "` Q" << static_cast<size_t>(a.qid) << '\n';
+    MixpanelQuestionAnsweredEvent ev(mixpanel_token_, a);
+    // WORKAROUND(sompylasar): `bricks::cerealize::JSON` cannot make more than one top-level key-value pair but we need this to build Mixpanel requests.
+    const std::string json = MultiKeyJSON(ev);
+    std::cerr << '@' << demo_id_ << " MixpanelUploader Event: " << json << std::endl;
+    const std::string base64_json = Base64Encode(json);
+    // WORKAROUND(sompylasar): Not using `https://`, could not send HTTPS request.
+    const std::string mixpanel_request = "http://api.mixpanel.com/track?data=" + base64_json;
+    std::cerr << '@' << demo_id_ << " MixpanelUploader Request: " << mixpanel_request << std::endl;
+    if (mixpanel_token_.empty()) {
```

Comment:
Yes, sure. But I could not make the `mixpanel_uploader_` optional in the `Controller` where the `Subscribe` happens.